### PR TITLE
Cuckoo Sendbox Authentication

### PR DESCRIPTION
**Pull request by MLainer1**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://github.com/demisto/etc/issues/19338)

## Description
added new authentication method, by api key.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CuckooSandbox/Integrations/integration-CuckooSandbox_README.md [hunk 1]
```diff
@@ -25,6 +25,7 @@
 <strong>Name</strong>: a textual name for the integration instance.</li>
 <li><strong>Server URL (e.g. <a href="https://192.168.0.1/" rel="nofollow">https://192.168.0.1</a>)</strong></li>
 <li><strong>Username (Only if your Cuckoo service requires HTTP auth)</strong></li>
+<li1>In order to authenticate with a Cluster Authentication Token, insert '__token' in the "Username" textbox, and the token itself in the "Password" textbox.</li1>
```
Comment (dantavori) [inline Packs/CuckooSandbox/Integrations/integration-CuckooSandbox_README.md:28]:
```suggestion
<li>In order to authenticate with an API token, insert '__token' in the "Username" textbox, and the token itself in the "Password" textbox.</li>
```

File: Packs/CuckooSandbox/ReleaseNotes/1_1_0.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#### Integrations
+##### Cuckoo Sandbox
+- Updated the authentication method to use api key.
```
Comment (dantavori) [inline Packs/CuckooSandbox/ReleaseNotes/1_1_0.md:3]:
```suggestion
- Added support for API key authentication.
```

File: Packs/CuckooSandbox/Integrations/integration-CuckooSandbox_README.md (hunk 1 above)
Comment (dantavori) [reply Packs/CuckooSandbox/Integrations/integration-CuckooSandbox_README.md:28]: let's add this line as an additionalinfo of the Username field in the yml (line 14)

File: Packs/CuckooSandbox/Integrations/integration-CuckooSandbox.yml [hunk 3]
```diff
@@ -148,51 +148,79 @@ script:
...
+                  Method: 'GET',
+                  Headers: {
+                      'Authorization': [full_api_key]
+                  },
+                  Username: username,
+                  Password: password
+              },
+              qParams['insecure'],
+              qParams['proxy']
+          );
```
Comment (dantavori) [inline Packs/CuckooSandbox/Integrations/integration-CuckooSandbox.yml:169]: is it possible to move this logic to a separate function `req()` and call it from every function that makes an API call?

**Review by dantavori — CHANGES_REQUESTED** [review]

File: Packs/CuckooSandbox/Integrations/integration-CuckooSandbox_README.md (hunk 1 above)
Comment (MLainer1) [reply Packs/CuckooSandbox/Integrations/integration-CuckooSandbox_README.md:28]: I've added the line in the description field in the yml

File: Packs/CuckooSandbox/Integrations/integration-CuckooSandbox_README.md [hunk 4]
```diff
@@ -25,12 +25,15 @@
...
 <li><strong>Username (Only if your Cuckoo service requires HTTP auth)</strong></li>
+<li>In order to authenticate with an API token, insert '__token' in the "Username" textbox, and the token itself in the "Password" textbox.</li>
 <li><strong>Trust any certificate (not secure)</strong></li>
 <li><strong>Use system proxy settings</strong></li>
 </ul>
 </li>
 <li>Click <strong>Test</strong> to validate the URLs, token, and connection.</li>
 </ol>
+<h2>Configuration params</h2>
+<p>NOTE: In order to authenticate with an API token, insert '__token' in the "Username" textbox, and the token itself in the "Password" textbox.</p>
```
Comment (dantavori) [inline Packs/CuckooSandbox/Integrations/integration-CuckooSandbox_README.md:36]: we agreed to put it in `CuckooSandbox_description.md`, right? if no such, we can add the "detaileddescription" field in the yml itself.
In addition, let's add there a short explanation for how to obtain the API key, if possible.

**Review by dantavori — CHANGES_REQUESTED** [review]

File: Packs/CuckooSandbox/Integrations/integration-CuckooSandbox.yml [hunk 5]
```diff
@@ -148,51 +148,70 @@ script:
         url = fixUrl(qParams['server']) + suffix;
         username = qParams['authentication'] ? qParams['authentication']['identifier'] : '';
         password = qParams['authentication'] ? qParams['authentication']['password'] : '';
+        headers = {};
+        if (username == '__token'){
+            username = '';
+            headers = {'Authorization': ['Bearer ' + password]};
```
[the remark this answers is not in the document]
Comment (MLainer1) [reply Packs/CuckooSandbox/Integrations/integration-CuckooSandbox.yml:154]: Somhow it does not work if you put password = '' in there, which is very strange

[the remark this answers is not in the document]
Comment (dantavori) [reply Packs/CuckooSandbox/Integrations/integration-CuckooSandbox.yml:154]: and if you don't pass the username & password at all, it does?

File: Packs/CuckooSandbox/Integrations/integration-CuckooSandbox_README.md (hunk 4 above)
Comment (MLainer1) [reply Packs/CuckooSandbox/Integrations/integration-CuckooSandbox_README.md:36]: I can't find that file, and I'll try to find information about the API key obtaining process

File: Packs/CuckooSandbox/Integrations/integration-CuckooSandbox.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (MLainer1) [reply Packs/CuckooSandbox/Integrations/integration-CuckooSandbox.yml:154]: I found the problem, the order of these statements was wrong. I fixed it :)

File: Packs/CuckooSandbox/Integrations/integration-CuckooSandbox.yml [hunk 6]
```diff
@@ -148,51 +149,72 @@ script:
...
-                Username: username,
-                Password: password
-            },
-            qParams['insecure'],
-            qParams['proxy']
-        );
+              url,
+              {
+                  Method: 'GET',
+                  Headers: headers,gco
```
Comment (dantavori) [inline Packs/CuckooSandbox/Integrations/integration-CuckooSandbox.yml:185]:
```suggestion
                  Headers: headers,
```

File: Packs/CuckooSandbox/Integrations/integration-CuckooSandbox.yml [hunk 7]
```diff
@@ -6,6 +6,7 @@ display: Cuckoo Sandbox
...
 description: Malware dynamic analysis sandboxing
+detaileddescription: In order to authenticate with an API token, insert '__token' in the "Username" textbox, and the token itself in the "Password" textbox. In new Cuckoo installations, a random token is automatically generated for you, and located in the cuckoo.conf file.
```
Comment (dantavori) [inline Packs/CuckooSandbox/Integrations/integration-CuckooSandbox.yml:9]:
```suggestion
detaileddescription: |-
   #### Authenticate with an API token
   In new Cuckoo installations, a random API token is automatically generated for you, and located in the cuckoo.conf file. In order to authenticate with an API token, insert '__token' in the "Username" textbox, and the token itself in the "Password" textbox.
```

**Review by dantavori — APPROVED** [review]

**Comment by MLainer1** [comment]
This pull request needs a force-merge due to a problem in a test playbook. V2 will be out in the next quarter

**Pull request merged**